fiftyone/core/dataset.py

```
@@ -11750,6 +12178,298 @@ def _get_media_type(sample):
+
+    kinds = set(kinds)
+    if len(kinds) > 1:
+        raise ValueError(
+            "A media-reference dataset cannot contain multiple reference "
+            "kinds"
+        )
+
+    return next(iter(kinds)) if kinds else None
+
+
+def _get_media_identity_mode(samples):
```

Ruff (0.16.2)

[warning] Too many return statements (8 > 6)

(PLR0911)